### PR TITLE
[TASK] Save server settings in blob (RT-3224)

### PR DESCRIPTION
**Pull request by darkdarkdragon**
Advanced: Save server settings in the blobvault

### CODE REVIEW

File: src/js/tabs/advanced.js [hunk 1]
```diff
@@ -178,6 +178,8 @@ AdvancedTab.prototype.angular = function(module)
         if (!store.disabled) {
           store.set('ripple_settings', JSON.stringify($scope.options));
         }
+        var servers = settings.getClearServers($scope.options.server.servers);
```
Comment (vhpoet) [inline src/js/tabs/advanced.js]: save and remove have 5 lines of the same code, can you move them to a separate method? let's say `.persist()`

File: src/js/util/settings.js [hunk 2]
```diff
@@ -60,3 +60,9 @@ exports.getSetting = function(userBlob, settingName, def) {
   }
   return def;
 };
+
+exports.getClearServers = function(servers) {
+  return _.map(servers, function(server) {
+    return _.pick(server, 'host', 'port', 'secure');
```
Comment (vhpoet) [inline src/js/util/settings.js]: What else could be in the server object at this point?

Comment (darkdarkdragon) [inline src/js/util/settings.js]: $$hashKey added by angular. Same object used as parameter to Remote, so url property get added by Server.
Also new servers added from interface has isEmptyServer property.

File: src/js/util/settings.js [hunk 3]
```diff
@@ -60,3 +60,11 @@ exports.getSetting = function(userBlob, settingName, def) {
   }
   return def;
 };
+
+exports.getClearServers = function(servers) {
+  return _.map(servers, function(server) {
+    var o = _.pick(server, 'host', 'port', 'secure');
+    o.port = +o.port;
```
Comment (jublo) [inline src/js/util/settings.js]: Maybe add a comment about this, or use the explicit variant, which might be easier to read.

**Comment by mrajvanshy** [comment]
@darkdarkdragon rebase

**Comment by vhpoet** [comment]
LGTM

**Comment by mrajvanshy** [comment]
LGTM

**Pull request merged**